src/zenml/client.py

```diff
@@ -2229,6 +2247,213 @@ def delete_stack_component(
+            container_registry_outputs = {
+                k: v
+                for k, v in outputs.items()
+                if k.startswith("container_registry")
+            }
+            if container_registry_outputs:
+                self.create_stack_component(
+                    name=container_registry_outputs["container_registry_name"],
+                    flavor=container_registry_outputs[
+                        "container_registry_flavor"
+                    ],
+                    component_type=StackComponentType.CONTAINER_REGISTRY,
+                    configuration=eval(
+                        container_registry_outputs[
+                            "container_registry_configuration"
+                        ]
+                    ),
+                )
+
+    def destroy_stack_component(
+        self,
+        ctx: click.Context,
+        name_id_or_prefix: Union[str, UUID],
+        component_type: StackComponentType,
+    ) -> None:
+        """Destroys a stack component.
+
+        Args:
+            ctx: The click context.
+            name: The name of the deployed stack component.
+            name_id_or_prefix: The model of the component to destroy.
+            component_type: The type of the stack component to destroy.
+        """
+        try:
+            component = self.get_stack_component(
+                name_id_or_prefix=name_id_or_prefix,
+                component_type=component_type,
+                allow_name_prefix_match=False,
+            )
+        except KeyError:
+            logger.info(
+                "Could not find a stack component with name or id %s",
+                name_id_or_prefix,
+            )
+            return
+
+        # if the component's metadata doesn't have a key created_by
+        # equal to 'recipe', then destroy cannot be called on it
+        if (
+            not component.metadata
+            or "created_by" not in component.metadata
+            or component.metadata["created_by"] != "recipe"
+        ):
+            logger.error(
+                "Cannot destroy stack component %s. It was not created by a "
+                "recipe.",
+                component.name,
+            )
+            return
+
+        from zenml.cli.stack_recipes import (
+            destroy,
+        )
+
+        STACK_COMPONENT_RECIPE_DIR = "deployed_stack_components"
+
+        # path should be fixed at a constant in the
+        # global config directory
+        path = Path(
+            os.path.join(
+                io_utils.get_global_config_directory(),
+                STACK_COMPONENT_RECIPE_DIR,
+            )
+        )
+
+        # set the stack component and flavor
+        component_flavor = {component_type.value: component.flavor}
+
+        # Invoke the destroy command
+        ctx.invoke(
+            destroy,
+            path=path,
+            stack_recipe_name=f"{component.metadata['cloud']}-modular",
+            **component_flavor,
+        )
```

Comment:
again, not sure this is the right behavior to be invoking CLI commands within the `client`